### ISSUE TITLE
Red Alert: Restore diplomacy changes and add optional switch to enable/disable

### ISSUE THREAD

**Issue by mikeik** [OpenRA/OpenRA#15061]
1. In earlier versions of openra, there was optional diplomacy. It was very basic - basically, at any time I can unally or ally with another human player
2. That player receives a notification that I have allied or unallied with them
3. This feature was removed with this change https://github.com/OpenRA/OpenRA/pull/10650
4. I regularly play this game with friends (THANK YOU for creating it, by the way - great to be able to play again), but for us one of the most fun things is to knock out the AI then turn on each other (we all play LAN)
5. So losing 'fragile diplomacy' was very sad, as we're stuck playing the last version that included it
6. This would be to put the option back in just for the Red Alert version, with a switch to enable/disable
7. If the switch is a hidden/command line/config file option, that would be totally fine
8. A nice-to-have would be to add in a notice to *all* players when an alliance state changes (e.g. Player 1 is now allied with Player 2, The alliance between Player 3 and Player 4 has ended)
9. I did check the bug tracker and I see this issue https://github.com/OpenRA/OpenRA/issues/14910 but I felt that I wanted to add the additional detail above and I didn't want to edit someone else's issue

Hopefully the above makes sense. If it does and it can be tagged appropriately, I would be happy to add a $50 bounty over in bountysource.

**Comment by KOYK** [comment]
I know I would love to use it again. ahh those where the days.

**Comment by mikeik** [comment]
Ok, I added a $50 bounty! Hopefully it could get done sometime :)

**Comment by BeanBnZ** [comment]
I recently made this issue and I was reported as a duplicate of this post,

Here it is below in support of adding alliances as an option in the map option menu.

Idea for TD/RA/D2k
Re introduce alliance option in map options menu, this will be especially useful in free for all matches & could be lot of fun in many situations. Like removing the build radius it won't always be used, but it will create new dimension of play.

**Comment by mikeik** [comment]
Hey @abcdefg30 , just wondering if this is something you have an interest in implementing in the future? Totally fine if the answer is no, I was just curious.

**Comment by BeanBnZ** [comment]
I have discussed this with a few people and it could be really useful and more importantly fun in ffa matches, especially when one person is almost dead due to 2v1 and they can become a vassal either temporary or permanent. 


An interesting idea would be to allow permanent alliances, and therefore creating teams in game until the end. I think this would be so much fun and addictive and create a whole new wave of ffa games.

**Comment by abcdefg30** [comment]
> Hey @abcdefg30 , just wondering if this is something you have an interest in implementing in the future?

Yes, interest certainly. I was sad when we had to remove the feature. Unfortunately, reimplementing it will not be trivial - if at all doable - and it is not on my ToDo list for the near future.

**Comment by mikeik** [comment]
> Its doable, just a huge pile of not very fun work.

Isn't that the way with most work :( For whatever its worth, I know the pain. I work at a software engineering company. 

Given that I can't code (unless you happen to write exclusively in sql queries...), is there anything that I can do to help?

**Comment by BeanBnZ** [comment]
What if we start a cash bounty for the Dev on this one? would that be acceptable?

**Comment by mikeik** [comment]
@BeanBnZ , I did include a cash bounty ($50) when I opened it. Maybe that's not enough to get someone excited, I don't know... :)

**Comment by BeanBnZ** [comment]
OK, I will match your bounty, Maybe we could get this bumped in the GitHub forum to see if anyone else is interested?

**Comment by matjaeck** [comment]
It's cool that you want to add some extra motivation to work on this feature, however not every dev is a fan of bounties. 

```
<pchote> foul_owl: i don't think bounties are a good idea
 <pchote> they are a PITA to deal with administratively (tax paperwork to extract money from the US), and bountysource are now charging penalty fees for people who don't extract money
```

**Comment by mikeik** [comment]
@matjaeck , that's interesting. Well, glad I wasted that $50, haha! :)

**Comment by mikeik** [comment]
@pchote , from what I remember, you can only retrieve it if $100+. 

@matjaeck , I was only kidding really. 

To be honest, if someone ever does complete this issue, they are very welcome to whatever they can get of the $50 after-tax :)

**Comment by BeanBnZ** [comment]
& they would also receive a further $50 from myself, much fun will be had with this feature.

**Comment by benhbell** [comment]
Adding to the bounty to match +50

**Comment by benhbell** [comment]
Just doing a little necromancy here, This is 100% still a subject of sadness when we play dune locally with my group. I would settle for even an MVP of `ally username on` and `ally username off` with no ui, just commands. 

I am also happy to help with graphics.

**Comment by mikeik** [comment]
I couldn't agree more Ben. I really want this! :-(

**Comment by ayancey** [comment]
$100 bounty from me. Just want the ability for me and a friend to control the same sets of units. In co-op for instance

Edit 2025-06-30: Still active :)

Edit 2026-08-29: still active!

**Comment by JamesRBDev** [comment]
I'll add my own $200 to the bounty. This is an important feature to me, and that brings the total bounty up to around $400.

**Comment by IamTomgamer** [comment]
If I had the money, I would add to the bounty. Also, I’m planning on playing this game with a friend of mine who really likes army/strategy games, so this would be a great addition!
